ibm/service/cis/resource_ibm_cis_dns_record.go

```diff
@@ -468,9 +468,13 @@ func ResourceIBMCISDnsRecordRead(d *schema.ResourceData, meta interface{}) error
 	}
 
 	d.Set(cisID, crn)
-	d.Set(cisDomainID, *result.Result.ZoneID)
+	if result.Result.ZoneID != nil {
+		d.Set(cisDomainID, *result.Result.ZoneID)
+	}
 	d.Set(cisDNSRecordID, *result.Result.ID)
-	d.Set(cisZoneName, *result.Result.ZoneName)
+	if result.Result.ZoneName != nil {
```

Comment:
Question, do you have to do the same thing in https://github.com/IBM-Cloud/terraform-provider-ibm/blob/18e89c7ff92c76d128b57b30aaecf19e4b8ad538/ibm/service/cis/data_source_ibm_cis_dns_records.go#L209